### ISSUE TITLE
Display the contents of all tabs to the window title

### ISSUE THREAD

**Issue by hleft** [qutebrowser/qutebrowser#7391]
I hope use `tabs.show never` and support display the contents of all tabs to the window title(then I can use window manager's statusbar instead browser's tab bar), and allow the current tab and other tabs to use different formats

This will beautiful and let the browser save the tab bar to display more content

**Comment by leo848** [comment]
how about instead setting `window.hide_decoration` to `True` to remove the title bar, so you can keep and configure your tab bar?

**Comment by hleft** [comment]
@leo848    I use window manager, the titlebar is hidden by default. In the case of non-fullscreen, there will always be a status bar, so I want to set the window title to display the content to the status bar. This will more beautiful because all applications share the same statusbar, and statusbar has more configuration options than browser tab bar

**Comment by leo848** [comment]
Could you maybe share a screenshot? I also use WMs from time to time and don't think I've ever seen one where I wanted this type of information to appear in the title bar.

**Comment by hleft** [comment]
@leo848  

![awesomewm's statusbar](https://user-images.githubusercontent.com/89069008/189665224-004a78ae-e3ab-4561-b37c-67b31c17c91a.png)

```
c.tabs.title.format = '{current_title} {scroll_pos} - qutebrowser'
c.window.title_format = '{current_title} {scroll_pos} - qutebrowser'
```

I hope make bar and browser have a different transparency, use a round bar or something

when at the top of the page, {scroll_pos} show top, I config delete `top - qutebrowser` in status bar , 

If title uses strange text, I can filter and change them. 

Although can setting the `c.tabs.title.format` and `c.window.title_format` to different format, but if make the content of all tabs available to the window manager will better

**Comment by leo848** [comment]
I agree, however I don't think an interactive interface (like a tab bar) belongs in an app's title. 

However I do think the tab title should be more adaptable via user-defined code (which is not possible afaik). There could be a function accepting either a function from string to string which modifies the existing tab title (as is the case with your WM, I presume) or a function from a context object holding the metadata to a string, building the tab title in the process (maybe this also deserves a new issue, I'm not sure about your original intention with this one).

**Comment by hleft** [comment]
@leo848  The original intention is to replace the browser's tab bar with the window's manager status bar. It's true that the tab bar shouldn't belong to the app's title, but if it were allowed, this problem would solve the problem perfectly..  (Even though qutebrowser supports the use of functions to modify tab content, I still prefer to use the status bar instead of the tab bar because status bar can be more customized)

**Comment by toofar** [comment]
I suspect this would be solved (or enabled at least) by #1578, despite that being about tab title it would probably apply to window titles too as they are the same implementation currently.

**Comment by The-Compiler** [comment]
Agreed - let's merge this into #1578.
